src/sagemaker_algorithm_toolkit/hyperparameter_validation.py

```diff
@@ -244,15 +244,26 @@ def _visit(name, visited, stack):
         return dependencies_stack
 
     def validate(self, user_hyperparameters):
-        # NOTE: 0. Validate required or fill in default.
+        # NOTE: 0. Convert possible aliases to original terms
+        tmp_user_hyperparameters = {}
+        hp_aliases = {"learning_rate": "eta", "min_split_loss": "gamma", "reg_lambda": "lambda", "reg_alpha": "alpha"}
+
+        for hp, value in user_hyperparameters.items():
+            if hp in hp_aliases:
+                tmp_user_hyperparameters[hp_aliases[hp]] = value
+            else:
+                tmp_user_hyperparameters[hp] = value
+        user_hyperparameters = tmp_user_hyperparameters
+
+        # NOTE: 1. Validate required or fill in default.
         for hp in self.hyperparameters:
             if hp not in user_hyperparameters:
                 if self.hyperparameters[hp].required:
                     raise exc.UserError("Missing required hyperparameter: {}".format(hp))
                 elif self.hyperparameters[hp].default is not None:
                     user_hyperparameters[hp] = self.hyperparameters[hp].default
 
-        # NOTE: 1. Convert hyperparameters.
+        # NOTE: 2. Convert hyperparameters.
         converted_hyperparameters = {}
         for hp, value in user_hyperparameters.items():
             try:
```

Comment:
With the above change, you could perhaps check for the alias here

Reply:
Yes, now with the above change, we can check for aliases here along with `# NOTE: 0`.

Reply:
sounds good.